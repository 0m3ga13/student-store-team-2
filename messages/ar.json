{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
