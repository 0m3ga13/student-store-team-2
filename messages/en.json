{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
