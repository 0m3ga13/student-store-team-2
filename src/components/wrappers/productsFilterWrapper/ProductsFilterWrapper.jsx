"use client";
import { useRouter, useSearchParams } from "next/navigation";
import { useTranslations } from "next-intl";
import React, { useEffect } from "react";
import { useState } from "react";
import { useRef } from "react";

import CategoryFilter from "@/components/filters/categoryFilter/CategoryFilter";
import NoteFilter from "@/components/filters/noteFilter/NoteFilter";
import PriceFilter from "@/components/filters/priceFilter/PriceFilter";
import StatusFilter from "@/components/filters/statusFilter/StatusFilter";

function ProductsFilterWrapper({
    autoSendRequest = true,
    toggleSidebar,
    categories,
}) {
    const searchParams = useSearchParams();
    const t = useTranslations("Index");
    const router = useRouter();
    const [price, setPrice] = useState({
        min: searchParams.get("minPrice") ? searchParams.get("minPrice") : 0,
        max: searchParams.get("maxPrice") ? searchParams.get("maxPrice") : 1000,
    });
    const [status, setStatus] = useState(
        searchParams.get("status") ? searchParams.get("status") : ""
    );
    const [note, setNote] = useState(
        searchParams.get("note") ? searchParams.get("note") : ""
    );
    const [category, setCategory] = useState(
        searchParams.get("category") ? searchParams.get("category") : ""
    );

    // track when the component is mounted
    const requestRef = useRef(false);
    // initial price
    const initialPrice = {
        min: 0,
        max: 1000,
    };

    const handlePriceChange = (obj) => {
        if (Object.keys(obj)[0] === "min") {
            setPrice({ ...price, min: obj.min });
        } else {
            setPrice({ ...price, max: obj.max });
        }
    };

    const handleStatusChange = (val) => {
        setStatus(val);
    };
    const handleNoteChange = (val) => {
        setNote(val);
    };
    const handleCategoryChange = (val) => {
        setCategory(val);
    };

    const handleSendRequest = () => {
        sendRequest();
        toggleSidebar();
    };
    const resetState = () => {
        setPrice(initialPrice);
        setStatus("");
        setNote("");
        setCategory("");
    };

    const sendRequest = () => {
        const queryParams = new URLSearchParams();

        const paramsArray = [
            { minPrice: price.min },
            { maxPrice: price.max === initialPrice.max ? "" : price.max },
            { status },
            { note },
            { category },
        ];
        paramsArray.forEach((param) => {
            if (param[Object.keys(param)[0]]) {
                queryParams.append(
                    Object.keys(param)[0],
                    param[Object.keys(param)[0]]
                );
            }
        });
        const url = searchParams.has("page")
            ? `page=${searchParams.get("page")}&${queryParams.toString()}`
            : queryParams.toString();

        router.push("/products" + "?" + url);
    };

    useEffect(() => {
        if (requestRef.current) {
            requestRef.current = false;
            return;
        }
        // Send a request each time one of the state variables changes
        if (autoSendRequest) {
            sendRequest();
        }
    }, [price.min, price.max, status, note, category]);
    return (
        <div className='max-sm:pb-12 '>
            <div className=' mb-6 '>
                <div className='flex w-full justify-between  mb-4 '>
                    <button
                        onClick={resetState}
                        className='block text-md py-1 px-3  bg-gray-200 rounded-sm hover:bg-gray-400 active:bg-gray-200'
                    >
                        {t("Reset")}
                    </button>
                    {/* show only in mobile view */}
                    {!autoSendRequest && (
                        <button
                            onClick={handleSendRequest}
                            className='block text-md py-1 px-3 text-white bg-accent rounded-sm hover:bg-gray-400 active:bg-gray-500'
                        >
                            {t("Filter")}
                        </button>
                    )}
                </div>
                <h1 className='block uppercase text-lg '>{t("Filter By")}: </h1>
            </div>
            <div className='flex flex-col gap-8'>
                <PriceFilter
                    minVal={price.min}
                    maxVal={price.max}
                    handlePriceChange={handlePriceChange}
                />
                <StatusFilter
                    handleStatusChange={handleStatusChange}
                    currentStatus={status}
                />
                <NoteFilter
                    currentNote={note}
                    handleNoteChange={handleNoteChange}
                />
                <CategoryFilter
                    categories={categories}
                    currentCategory={category}
                    handleCategoryChange={handleCategoryChange}
                />
            </div>
        </div>
    );
}

export default ProductsFilterWrapper;
