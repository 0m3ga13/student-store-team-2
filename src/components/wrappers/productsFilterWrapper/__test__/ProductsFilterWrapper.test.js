import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import messages from "@/../messages/en.json";

import ProductsFilterWrapper from "../ProductsFilterWrapper";

// Mock the "next/navigation" module
jest.mock("next/navigation", () => ({
    useRouter: jest.fn(() => ({
        push: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({
        get: jest.fn(),
    })),
    usePathname: jest.fn(),
    useParams: () => ({
        locale: "en",
    }),
}));
it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <ProductsFilterWrapper />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
