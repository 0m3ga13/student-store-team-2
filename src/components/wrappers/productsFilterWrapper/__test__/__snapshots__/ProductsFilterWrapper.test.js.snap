// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className=" mb-6"
  >
    <button
      className="block text-md py-1 px-3 ml-auto mb-4 w-fit bg-gray-200 rounded-sm hover:bg-gray-400 active:bg-gray-200"
      onClick={[Function]}
    >
      Reset
    </button>
    <h1
      className="block uppercase text-lg "
    >
      filter by: 
    </h1>
  </div>
  <div
    className="flex flex-col gap-8"
  >
    <div
      className="pb-10"
    >
      <h2
        className="font-jost sm:text-lg"
      >
        Price Filter
      </h2>
      <div
        className="mt-4"
      >
        <input
          className="thumb thumb--left pointer-events-none absolute h-0 w-52 outline-none z-30"
          max={1000}
          min={0}
          onChange={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "zIndex": false,
            }
          }
          type="range"
          value={0}
        />
        <input
          className="thumb thumb--right pointer-events-none absolute h-0 w-52 outline-none z-40 "
          max={1000}
          min={0}
          onChange={[Function]}
          onMouseUp={[Function]}
          type="range"
          value={1000}
        />
        <div
          className="relative w-52"
        >
          <div
            className="slider__track absolute rounded h-1 bg-gray-200 w-full z-10"
          />
          <div
            className="slider__range absolute rounded h-1 z-20 bg-[#821F40]"
          />
          <div
            className="slider__left-value absolute text-[#55585B] text-xs left-2 mt-5"
          >
            0
          </div>
          <div
            className="slider__right-value absolute text-[#55585B] text-xs mt-5 right-1"
          >
            1000
          </div>
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <h2
        className="capitalize text-lg"
      >
        product status
      </h2>
      <div
        className="mt-4 border-t border-t-[#EEEEEE] flex flex-col gap-5 py-4 px-3"
      >
        <label
          className="group relative text-[#55585B] text-sm "
        >
          For Sale
          <input
            className="absolute opacity-0 cursor-pointer peer"
            name="radio"
            onChange={[Function]}
            type="radio"
            value="sale"
          />
          <span
            className="absolute -top-2 -left-3 rounded-sm h-8 w-28 group-hover:bg-gray-200 -z-10   "
          />
        </label>
        <label
          className="group relative text-[#55585B] text-sm "
        >
          For Barrow
          <input
            className="absolute opacity-0 cursor-pointer peer"
            name="radio"
            onChange={[Function]}
            type="radio"
            value="barrow"
          />
          <span
            className="absolute -top-2 -left-3 rounded-sm h-8 w-28 group-hover:bg-gray-200 -z-10   "
          />
        </label>
      </div>
    </div>
    <div>
      <h2
        className="capitalize text-lg"
      >
        product note
      </h2>
      <div
        className="flex gap-2 mt-4 border-t border-t-[#EEEEEE] py-4"
      >
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={5}
        >
          5
        </button>
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={6}
        >
          6
        </button>
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={7}
        >
          7
        </button>
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={8}
        >
          8
        </button>
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={9}
        >
          9
        </button>
        <button
          className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
          onClick={[Function]}
          value={10}
        >
          10
        </button>
      </div>
    </div>
    <div
      className=""
    >
      <h2
        className="capitalize text-lg"
      >
        categories
      </h2>
      <div
        className="mt-4 border-t border-t-[#EEEEEE] flex flex-col items-start gap-5 py-4 px-3"
      />
    </div>
  </div>
</div>
`;
